carbon-design-system/carbon
# feat(component): replaced `SideNavMenuItem.js` -> `SideNavMenuItem.tsx` component

Closes #13607

To add a new file `SideNavMenuItem.tsx` with it's corresponding types.

#### Changelog

**Changed**

- Updated TypeScript types for the SideNavMenuItem component to resolve type error.

#### Testing / Reviewing

To verify the changes:

1. Ensure that the SideNavMenuItem component renders without any TypeScript errors.
2. Test the component with different props and verify that it behaves as expected.
3. Verify that the component works correctly in various usage scenarios, including cases with and without the isActive prop.
4. Check if the component integrates seamlessly with the rest of the application's components and features.
5. Run existing tests to ensure that the changes do not break any existing functionality.
